src/main/java/bio/terra/cli/context/utils/Logger.java

```diff
@@ -0,0 +1,184 @@
+package bio.terra.cli.context.utils;
+
+import static ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME;
+
+import bio.terra.cli.context.GlobalContext;
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.OutputStreamAppender;
+import ch.qos.logback.core.filter.Filter;
+import ch.qos.logback.core.rolling.FixedWindowRollingPolicy;
+import ch.qos.logback.core.rolling.RollingFileAppender;
+import ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy;
+import ch.qos.logback.core.spi.FilterReply;
+import ch.qos.logback.core.util.FileSize;
+import ch.qos.logback.core.util.StatusPrinter;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.File;
+import org.slf4j.LoggerFactory;
+
+/** Utility methods for logging. */
+public class Logger {
+  private static final String LOG_FORMAT =
+      "%d{HH:mm:ss.SSS} [%thread] %-5level %logger{50} - %msg%n";
+
+  private static final long MAX_FILE_SIZE = 50 * FileSize.MB_COEFFICIENT; // 50 MB
+  private static final int MAX_NUM_FILES = 5;
+
+  private final GlobalContext globalContext;
+
+  /**
+   * Constructor for class that sets up logging.
+   *
+   * @param globalContext the global context object
+   */
+  public Logger(GlobalContext globalContext) {
+    this.globalContext = globalContext;
+  }
+
+  /**
+   * Setup a file and console appender for the root logger. Each may use a different logging level,
+   * as specified in the global context.
+   */
+  @SuppressFBWarnings(
+      value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+      justification =
+          "An NPE would only happen here if there was an error getting the global context directory,"
+              + " and an exception would have been thrown earlier when we first read in the global "
+              + "context file anyway.")
+  public void setupLogging() {
+    LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+    loggerContext.reset();
+
+    // turn on logging for the logger for errors only
+    StatusPrinter.printIfErrorsOccured(loggerContext);
+
```

Comment:
Thanks for adding the `RollingFileAppender`!

Reply:
Thank you for the great suggestion! I'm working on addressing Greg's comments now, and then I think I'm going to ping you both for a re-review, if you don't mind. Enough has changed since the original that another once-over by you guys would be much appreciated.